src/libYARP_dev/src/yarp/dev/Map2DPath.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2006-2019 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+
+#include <yarp/dev/Map2DPath.h>
+#include <yarp/os/ConnectionReader.h>
+#include <yarp/os/ConnectionWriter.h>
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+#include <cassert>
+#include <functional>
+
+using namespace yarp::dev;
+using namespace yarp::dev::Nav2D;
+using namespace yarp::sig;
+using namespace yarp::os;
+using namespace yarp::math;
+using namespace std;
+
+bool Map2DPath::operator!=(const Map2DPath& r) const
+{
+    for (size_t i = 0; i < waypoints.size(); i++)
+        if (this->waypoints[i].map_id == r.waypoints[i].map_id ||
+            this->waypoints[i].x == r.waypoints[i].x ||
+            this->waypoints[i].y == r.waypoints[i].y ||
+            this->waypoints[i].theta == r.waypoints[i].theta) return false;
+    return true;
+}
+
+bool Map2DPath::operator==(const Map2DPath& r) const
+{
+    for (size_t i = 0; i < waypoints.size(); i++)
+        if (this->waypoints[i].map_id != r.waypoints[i].map_id ||
+            this->waypoints[i].x != r.waypoints[i].x ||
+            this->waypoints[i].y != r.waypoints[i].y ||
+            this->waypoints[i].theta != r.waypoints[i].theta) return false;
+    return true;
+}
+
+std::string Map2DPath::toString() const
+{
+    std::ostringstream stringStream;
+    stringStream.precision(-1);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Invalid stringStream.precision() argument nr 1. The value is -1 but the valid values are '0:'.](https://app.codacy.com/app/robotology/yarp/pullRequest?prid=4615567)